### PR TITLE
feat(FN-2405): add validation for exchange rates

### PR DESCRIPTION
**Pull request by francescastocco**
## Introduction :pencil2:
We need additional validation if portal-ui for payment exchange rate & accrual exchange rate

## Resolution :heavy_check_mark:
- Validate exchange rate value is 1 or null if currencies match
- Validate exchange rate value even if 2nd currency is `null`
- Updated unit tests

## Miscellaneous :heavy_plus_sign:
- CLAMAV_SCANNING_ENABLED shouldn't default to true now, don't know if we want this?
- Merging into e2etf-sql instead of main for release 2, new PR [here](https://github.com/UK-Export-Finance/dtfs2/pull/2916)

### CODE REVIEW

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-payment-exchange-rate-error.test.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ describe('generatePaymentExchangeRateError', () => {
         row: 1,
       },
       'payment exchange rate': {
-        value: 'abc',
+        value: null,
```
Comment (JoshBinns2000) [inline portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-payment-exchange-rate-error.test.js:53]: Are we sure that the csv parser will turn this to `null` if the cell is empty? I haven't looked too much into it, but I could see the values for these things being `null`, `undefined` or `''` when parsed 🤷

Comment (francescastocco) [reply portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-payment-exchange-rate-error.test.js:53]: Just tested, it's parsing empty cells as `null`

File: docker-compose.yml [hunk 2]
```diff
@@ -161,7 +161,7 @@ services:
       - CLAMAV_PORT=3310
       - CLAMAV_HOST=clamav
       - CLAMAV_DEBUG_MODE_ENABLED=true
-      - CLAMAV_SCANNING_ENABLED=true
+      - CLAMAV_SCANNING_ENABLED
```
Comment (ChristianMcc) [inline docker-compose.yml:164]: definitely a good change!

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-total-fees-accrued-exchange-rate-error.test.js [hunk 3]
```diff
@@ -3,7 +3,31 @@ const { FILE_UPLOAD } = require('../../../../constants/file-upload');
...
+        column: 3,
+      },
+    };
+
+    const exchangeRateError = generateTotalFeesAccruedExchangeRateError(csvDataRow);
+
+    expect(exchangeRateError).toEqual(null);
+  });
+  
+  it('returns null if accrual currency is null and accrual exchange rate is not a number', async () => {
```
Comment (ChristianMcc) [inline portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-total-fees-accrued-exchange-rate-error.test.js:30]: returns an error*

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-total-fees-accrued-exchange-rate-error.test.js [hunk 4]
```diff
@@ -22,12 +46,81 @@ describe('generateTotalFeesAccruedExchangeRateError', () => {
...
+      value: 'abc',
+      exporter: testExporterName,
+    };
+
+    const exchangeRateError = generateTotalFeesAccruedExchangeRateError(csvDataRow);
+
+    expect(exchangeRateError).toEqual(expectedError);
+  });
+  
+  it('returns null if accrual currency is null and accrual exchange rate is too long', async () => {
```
Comment (ChristianMcc) [inline portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-total-fees-accrued-exchange-rate-error.test.js:62]: returns an error*

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-total-fees-accrued-exchange-rate-error.test.js [hunk 5]
```diff
@@ -22,12 +46,81 @@ describe('generateTotalFeesAccruedExchangeRateError', () => {
...
+      value: '1.738491847362543',
+      exporter: testExporterName,
+    };
+
+    const exchangeRateError = generateTotalFeesAccruedExchangeRateError(csvDataRow);
+
+    expect(exchangeRateError).toEqual(expectedError);
+  });
+
+  it('returns null if accrual currency is the same as base currency and accrual exchange rate is null', async () => {
```
Comment (ChristianMcc) [inline portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-total-fees-accrued-exchange-rate-error.test.js:94]: returns an error*

**Review by ChristianMcc — COMMENTED** [review]
nice, LGTM :) just a couple of copy paste issues to fix on test names

**Review by ChristianMcc — COMMENTED** [review]
Just remembered after finishing the review - can we add some additional validation to check that if the two currencies are equal and the payment exchange rate isn't null (the current check you've added) that the payment exchange rate is equal to 1 please?

File: docker-compose.yml (hunk 2 above)
Comment (francescastocco) [reply docker-compose.yml:164]: I've updated the other docker-compose files too

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-total-fees-accrued-exchange-rate-error.test.js (hunk 5 above)
Comment (francescastocco) [reply portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-cell-validators/generate-total-fees-accrued-exchange-rate-error.test.js:94]: I think this one should be null

**Comment by francescastocco** [comment]
New PR for release 2 https://github.com/UK-Export-Finance/dtfs2/pull/2916

**Pull request closed without merging**